### ISSUE TITLE
Only set meta data before TDySetFromOptions

### ISSUE THREAD

**Issue by bishtgautam** [TDycores-Project/TDycore#107]
- [ ] Move creation of DM
- [ ] Move allocation of memory

**Comment by jeff-cohere** [comment]
I would go so far as to say "only set metadata before TDySetup".

**Comment by ghammond86** [comment]
Current TDycore Order of Operations
```TDyCreate - creates and initializes the TDy struct.  No dynamic memory allocation
  TDyIOCreate
TDySetMode - sets flow mode
TDySetDiscretizationMethod - sets the discretization method
TDySet<XYZ>Function (e.g. TDySetPermeabilityFunction) - can be called after after TDy struct is created
TDySetFromOptions - harvests the metadata and overrides defaults, except for DM metadata
TDyCreateDM - can occur anywhere prior to TDySetupDiscretization
TDyDistributeDM - converts to parallel
TDySetupDiscretization - sets DM pointer, calculates grid cells, and allocates parameters arrays in TDy struct
  TDyAllocate - allocates memory for parameter arrays
Function that set values must be called or set after TDyAllocate (e.g. TDySetPermeabilityScalar, TDySetPermeabilityDiagonal, etc)
TDySetup
  TDySetupDiscretizationMethod - initializes numerical method (MPFA_O,BDM, WY, TPF)
  sets up regression linkage
```

Proposed TDycore Order of Operations
```
TDyCreate
TDySet<A>
TDySet<D>
TDySet<B>
TDySet<C>
TDySetDM
TDySetSeepageBC
TDySetFromOptions
  TDyCreateGrid
    if (!tdy->dm)
      TDyCreateDM
      DMSetUseNatural
      TDyDistributeDM
    TDyMalloc
    TDyRegressionInitialize
Set properties for each cell e.g. TDySetBlockPermeabilityValuesLocal. These call can ONLY occur after TDySetFromOptions
TDySetupNumericalMethod
  If permeability function and permeability values are not set, use default permeability value
  GMatrix, Transmissiblity matrix
TDyRunToTime
```

**Comment by bishtgautam** [comment]
Fixed by #109
